pkg/provisioner/controller.go

```diff
@@ -200,28 +200,23 @@ func (c *obcController) processNextItemInQueue() bool {
 func (c *obcController) syncHandler(key string) error {
 
 	setLoggersWithRequest(key)
-	logD.Info("new Reconcile iteration")
+	logD.Info("reconciling claim")
 
 	obc, err := claimForKey(key, c.libClientset)
 	if err != nil {
-		return fmt.Errorf("request key %q: %v", key, err)
-	}
-	if obc == nil {
-		return fmt.Errorf("unexpected nil obc for request key %q", key)
+		if errors.IsNotFound(err) {
+			log.Info("OBC vanished, assuming it was deleted")
+			return nil
```

Comment:
In this scenario will the lib clean up the OB, secret, and CM if they still exist?

Reply:
Right.  If the OBC vanished, then it must not have had a finalizer.  And if there was no finalizer, then never made it to the beginning of the sync, and so no resources would have been created.

+1 for comment, will add